component/client-handler/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/jwt/PrivateKeyJWTClientAuthHandler.java

```diff
@@ -111,15 +105,21 @@ private boolean isValidJWTClientAssertionRequest(String clientAssertionType, Str
     }
 
     @Override
-    public boolean authenticateClient(OAuthTokenReqMessageContext tokReqMsgCtx)
-            throws IdentityOAuth2Exception {
+    public boolean authenticateClient(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2Exception {
+        boolean isValid = false;
         try {
             SignedJWT signedJWT = getSignedJWT(tokReqMsgCtx);
-            return jwtValidator.isValidToken(signedJWT);
+            isValid = jwtValidator.isValidToken(signedJWT);
```

Comment:
if isValid is false can return at this point

Reply:
+1, Fixed.